src/s3path.jl

```diff
@@ -107,6 +108,17 @@ function S3Path(
     )
 end
 
+function S3Path(path::S3Path; version::AbstractS3Version)
+    if !isnothing(path.version)
+        throw(
+            ArgumentError(
+                "Can't construct versioned path, input already specifies version: $path"
```

Comment:
Why not allow the keyword version to overwrite the version of the `S3Path` silently? Should probably document the rational for this as part of the docstring